src/snowflake/snowpark/modin/pandas/window.py

```diff
@@ -121,7 +121,7 @@ def __init__(
     ) -> None:
         # TODO: SNOW-1063358: Modin upgrade - modin.pandas.window.Rolling
         # Raise ValueError when invalid parameter values/combinations
-        if (isinstance(window, int) and window <= 0) or window is None:
+        if (isinstance(window, int) and window < 0) or window is None:
```

Comment:
+1. Would you open an accompanying issue in Modin and link here?

Reply:
@sfc-gh-dpetersohn This isn't a Modin issue (Modin handles these cases correctly), but Modin just doesn't do this error checking in the frontend layer: https://github.com/modin-project/modin/blob/main/modin/pandas/window.py#L99. I added these checks at the frontend for Snowpark pandas, should I file this as a feature request or leave it be?

Reply:
Your change is about expanding, could you move things related to rolling to a different PR. If that is just a case that Snowpark pandas does not handle yet, I don't think it should stay at frontend layer (the error message also doesn't match the check), should be at the snowpark pandas backend layer. Also, what happens when rolling is 0, is it just require a different implementation to handle it?

Reply:
Rolling/expanding are very similar (expanding is cumulative and rolling is not) so I don't think it needs two separate PRs as I'm also implementing rolling.count in this PR (it's very similar implementation to expanding.count).

The error message at the frontend layer matches the errors that pandas/Modin throw as well when rolling/expanding are called with invalid parameters, so I think having it at the frontend layer is a good thing/beneficial.

Rolling is 0 is a special case, I haven't specifically looked into how to implement it but it does not work with the current implementation. I'm sure the current implementation is extensible but out of scope for this PR.

Reply:
The are similar, but not the same, and doesn't seem matching the purpose of your current pr, your current pr intended to add support for new function, and make the review little bit difficult. for example, if you put the change here, i would expect you also add test to check this new case also, but seems not.

Also as where the check should happen, snowpark pandas specific check should happen at backend layer, or the overwritten layer, this folder is target for removed. From what you have described, when min_period = 0, it works for modin, it just doesn't work for Snowpark pandas, and we can potentially extend our backend to handle it. Then at backend should be fine, and i belive you already did the check at backend

Reply:
This isn't a Snowpark pandas specific check (that happens in the backend), this is a pandas check (all pandas cases where window < 0 results in this error).

The error message is `ValueError("window must be an integer 0 or greater")` the `=` part of the comparison was just a typo being fixed here.

Reply:
I added the window=0 negative test (checking it throws a NotImplementedError) to `test_rolling_params_unsupported`

Reply:
i see, i might read the message in an opposite way, then I think we should open an modin issue to upstreaming the frontend check changes.

For the test, there seems a kind of behavior change here, before this case raises a ValueError, now it start throwing an not implemented error.

Also, please add this change to CHAGNELOG for the window = 0 case

Reply:
Added it as a bug fix: `Fixed a bug in `DataFrame.rolling` and `Series.rolling` so `window=0` now throws `NotImplementedError` instead of `ValueError``